### ISSUE TITLE
[RFE] cmd/kubectl-version: Add more information to `version` command

### ISSUE THREAD

**Issue by blanquicet** [inspektor-gadget/inspektor-gadget#1060]
## Current situation

With the new information added to the `version` command, it is now possible to be sure we are running the same version on the client and server side:

```bash
$ ./kubectl-gadget version
Client version: v0.10.0
Server version: v0.10.0
```

However, during development, we get this:

```bash
$ ./kubectl-gadget version
Client version: v0.8.0-55-g4289e7627b24-dirty
Server version: my-branch-name
```

We cannot be sure we are running the same version with such information.

## Ideal future situation

As discussed in https://github.com/inspektor-gadget/inspektor-gadget/pull/977#discussion_r997982715, we could add more information as `kubectl` does. 

The question is, what information can we add to ensure we are using the same version on both sides?

**Comment by blanquicet** [comment]
> > We cannot be sure we are running the same version with such information.
> 
> I don't see how we can improve it for development scenarios. In the end it's the developers that knows what they're running. We could include some information about the tag and so on, but IMO if you use a custom image during development it's up to you to be sure you're using the right version. What do you think?

During development, I create the `add-new-feature`, commit the new feature and build both the client and server. The versions will be something like after deployment:

```bash
Client version: v0.24.0-168-g8298c7884008
Server version: add-new-feature
```

Where `8298c7884008` is the last commit I did. Then, I continue the development and commit another change `4289e7627b24`. Re-build everything, deploy it and now I have:

```bash
Client version: v0.24.0-168-g4289e7627b24
Server version: add-new-feature
```

How do I know that the sever side was correct updated and I'm not using `8298c7884008`? That's where I think we could help developer and at least include the commit SHA when it is not a tag. WDYT?

**Comment by mauriciovasquezbernal** [comment]
Makes sense. I think it's very easy to do once the version package proposed in https://github.com/inspektor-gadget/inspektor-gadget/issues/1795#issuecomment-1924266373 is introduced. Then we won't need to rely on the image tag in https://github.com/inspektor-gadget/inspektor-gadget/blob/fd1702ec3451c044c10ade79eba7916ec3e9b451/cmd/kubectl-gadget/version.go#L59-L60

but print the version that includes the commit
